.github/workflows/build-push.yml

```diff
@@ -0,0 +1,40 @@
+name: Build + Push to Azure
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Source branch to deploy"
+        required: false
+        default: "main"
+        type: string
+      environment:
+        description: "Determines where build gets pushed to"
+        required: false
+        default: "preview"
+        type: choice
+        options:
+          - preview
+          - dev
+          - staging
+          - prod
+  push:
+    branches:
+      - main
```

Comment:
Similar to the comment above, I'm assuming this is going to publish our public docs for every push to main? if so, we don't want that behavior, we only publish from our latest release branch.